### PR TITLE
Fix validation of dynamic Enum inside a Tuple

### PR DESCRIPTION
**Pull request by mdickinson**
This PR fixes validation of dynamic enumerations inside a Tuple. Previously, for the dynamic case of an `Enum`, no validation was performed until trait-set time. This PR changes the logic so that the validate method (which the Tuple uses for its item validation) is no longer None.

This fixes the issue originally reported in #1385. There are other issues identified in the #1385 comments that still need to be fixed (or have new issues opened) before #1385 can be closed.

**Checklist**
- [x] Tests
- [ ] Update API reference (`docs/source/traits_api_reference`)  N/A
- [ ] Update User manual (`docs/source/traits_user_manual`)  N/A
- [ ] Update type annotation hints in `traits-stubs`  N/A

### CODE REVIEW

File: traits/trait_types.py [hunk 1]
```diff
@@ -2074,10 +2075,16 @@ def _get(self, object, name, trait):
         return value
 
     def _set(self, object, name, value):
-        """ Sets the current value of a dynamic range trait.
+        """ Sets the current value of a dynamic enum trait.
```
Comment (mdickinson) [inline traits/trait_types.py:2078]: Unrelated drive-by fix in this line, fixing copypasta ("range" -> "enum")

**Review by kitchoi — APPROVED** [review]
LGTM

**Comment by mdickinson** [comment]
Thanks @kitchoi for reviewing. I added the new test (for a dynamic `Enum` inside a `List`) back in; it was failing before because I messed up the trait names. (Possibly a reason to use `HasStrictTraits` instead of `HasTraits` even when writing tests.)

**Comment by kitchoi** [comment]
Thank you! Still LGTM. So this PR can close #1385 then?

**Comment by mdickinson** [comment]
> So this PR can close #1385 then?

Yes, I guess so. I've opened an issue for the `Tuple` exception-silencing (#1389). I need to open an issue, or create a fix, for the corresponding issue in `BaseRange`.
